Sources/XCTest/XCTestRun.swift

```diff
@@ -0,0 +1,150 @@
+    }
+
+    /// `true` if all tests in the run completed their execution without
+    /// recording any failures, otherwise `false`.
+    public var hasSucceeded: Bool {
+        guard stopped else {
+            return false
+        }
+        return totalFailureCount == 0
+    }
+
+    /// Designated initializer for the XCTestRun class.
+    /// - Parameter test: An XCTest instance.
+    /// - Returns: A test run for the provided test.
+    public required init(test: XCTest) {
+        self.test = test
+    }
+
+    /// Start a test run. Must not be called more than once.
+    public func start() {
+        guard !started else {
+            fatalError("Invalid attempt to start a test run that has " +
+                       "already been started: \(self)")
+        }
+        guard !stopped else {
+            fatalError("Invalid attempt to start a test run that has " +
+                       "already been stopped: \(self)")
+        }
+
+        startDate = NSDate()
+    }
+
+    /// Stop a test run. Must not be called unless the run has been started.
+    /// Must not be called more than once.
+    public func stop() {
+        guard started else {
+            fatalError("Invalid attempt to stop a test run that has " +
+                       "not yet been started: \(self)")
+        }
+        guard !stopped else {
+            fatalError("Invalid attempt to stop a test run that has " +
+                       "already been stopped: \(self)")
+        }
+
+        executionCount += 1
+        stopDate = NSDate()
+    }
+
+    /// Records a failure in the execution of the test for this test run. Must
+    /// not be called unless the run has been started. Must not be called if the
+    /// test run has been stopped.
+    /// - Parameter description: The description of the failure being reported.
+    /// - Parameter filePath: The file path to the source file where the failure
+    ///   being reported was encountered or nil if unknown.
+    /// - Parameter lineNumber: The line number in the source file at filePath
+    ///   where the failure being reported was encountered.
+    /// - Parameter expected: `true` if the failure being reported was the
+    ///   result of a failed assertion, `false` if it was the result of an
+    ///   uncaught exception.
+    public func recordFailureWithDescription(description: String, inFile filePath: String, atLine lineNumber: UInt, expected: Bool) {
+        guard started else {
+            fatalError("Invalid attempt to record a failure for a test run " +
+                       "that has not yet been started: \(self)")
+        }
+        guard !stopped else {
+            fatalError("Invalid attempt to record a failure for a test run " +
+                       "that has already been stopped: \(self)")
+        }
+
+        if expected {
+            failureCount += 1
+        } else {
+            unexpectedExceptionCount += 1
+        }
+    }
+
+    private var started: Bool {
```

Comment:
Done! Also, I think I found a Swift bug. After changed these to `isStarted` and `isStopped`, I tried rebuilding the project and it failed:

```
d /Users/bgesiak/Library/Developer/Xcode/DerivedData/XCTest-gdiiwlsynrssmkdgamrgfoqbjewl/Build/Products/Debug/SwiftXCTest.framework/Versions/A/SwiftXCTest normal x86_64
    cd /Users/bgesiak/GitHub/apple/swift-corelibs-xctest
    export MACOSX_DEPLOYMENT_TARGET=10.11
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -arch x86_64 -dynamiclib -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk -L/Users/bgesiak/Library/Developer/Xcode/DerivedData/XCTest-gdiiwlsynrssmkdgamrgfoqbjewl/Build/Products/Debug -F/Users/bgesiak/Library/Developer/Xcode/DerivedData/XCTest-gdiiwlsynrssmkdgamrgfoqbjewl/Build/Products/Debug -filelist /Users/bgesiak/Library/Developer/Xcode/DerivedData/XCTest-gdiiwlsynrssmkdgamrgfoqbjewl/Build/Intermediates/XCTest.build/Debug/SwiftXCTest.build/Objects-normal/x86_64/SwiftXCTest.LinkFileList -install_name @rpath/SwiftXCTest.framework/Versions/A/SwiftXCTest -Xlinker -rpath -Xlinker @executable_path/../Frameworks -Xlinker -rpath -Xlinker @loader_path/Frameworks -mmacosx-version-min=10.11 -Xlinker -no_deduplicate -fobjc-link-runtime -L/Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2016-03-24-a.xctoolchain/usr/lib/swift/macosx -Xlinker -add_ast_path -Xlinker /Users/bgesiak/Library/Developer/Xcode/DerivedData/XCTest-gdiiwlsynrssmkdgamrgfoqbjewl/Build/Intermediates/XCTest.build/Debug/SwiftXCTest.build/Objects-normal/x86_64/SwiftXCTest.swiftmodule -framework SwiftFoundation -single_module -compatibility_version 1 -current_version 1 -Xlinker -dependency_info -Xlinker /Users/bgesiak/Library/Developer/Xcode/DerivedData/XCTest-gdiiwlsynrssmkdgamrgfoqbjewl/Build/Intermediates/XCTest.build/Debug/SwiftXCTest.build/Objects-normal/x86_64/SwiftXCTest_dependency_info.dat -o /Users/bgesiak/Library/Developer/Xcode/DerivedData/XCTest-gdiiwlsynrssmkdgamrgfoqbjewl/Build/Products/Debug/SwiftXCTest.framework/Versions/A/SwiftXCTest

Undefined symbols for architecture x86_64:
  "SwiftXCTest.XCTestRun.(started in _591E1E8C99D08309B8F88D133866F0A8).getter : Swift.Bool", referenced from:
      type metadata for SwiftXCTest.XCTestCaseRun in XCTestCaseRun.o
      type metadata for SwiftXCTest.XCTestSuiteRun in XCTestSuiteRun.o
  "SwiftXCTest.XCTestRun.(stopped in _591E1E8C99D08309B8F88D133866F0A8).getter : Swift.Bool", referenced from:
      type metadata for SwiftXCTest.XCTestCaseRun in XCTestCaseRun.o
      type metadata for SwiftXCTest.XCTestSuiteRun in XCTestSuiteRun.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Clearing out derived data solved the problem. Looks like incremental builds are kind of broken when only modifying private members?